lexos/models/content_analysis_model.py

```diff
@@ -0,0 +1,481 @@
+        for index, (score, formula, file) in enumerate(zip(self.scores,
+                                                           self._formulas,
+                                                           self._corpus)):
+            scores_sum += score
+            total_word_counts_sum += \
+                file.total_word_count
+            formulas_sum += formula
+        if len(self.scores) != 0:
+            scores_avg = round(
+                (float(scores_sum) / len(self.scores)), ndigits=3)
+        else:
+            scores_avg = 0
+        if len(self._corpus) != 0:
+            average = (float(total_word_counts_sum) / (len(self._corpus)))
+            total_word_counts_avg = round(average, ndigits=1)
+        else:
+            total_word_counts_avg = 0
+        if len(self._formulas) != 0:
+            sums_avg = round((float(formulas_sum) / len(self._formulas)),
+                             ndigits=1)
+        else:
+            sums_avg = 0
+        self._averages.append("Averages")
+        for dict_index, _ in enumerate(active_dicts):
+            cat_count = sum([counter[dict_index]
+                             for counter in self._counters])
+            if len(self._counters) != 0:
+                self._averages.append(round(
+                    float(cat_count) / len(self._counters), ndigits=1))
+            else:
+                self._averages.append(0)
+        self._averages.append(sums_avg)
+        self._averages.append(total_word_counts_avg)
+        self._averages.append(scores_avg)
+
+    def join_active_dicts(self) -> list:
+        """Joins all active dicts into on list of Phrase objects.
+
+        :return: list of phrases contained in the active dictionaries
+        """
+        active_dicts = self.get_active_dicts()
+        dictionaries = [Phrase(content=phrase, dict_label=dictionary.label)
+                        for dictionary in active_dicts
+                        for phrase in dictionary.content if phrase != '']
+        dictionaries.sort(key=lambda x: len(x.content.split()), reverse=True)
+        return dictionaries
+
+    def to_html(self) -> str:
+        """Generates an html table from dataframe.
+
+        :return: dataframe table in html
+        """
+        df = self.to_data_frame()
+        html = df.to_html(classes="table table-striped table-bordered",
+                          index=False)
+        return html
+
+    def to_data_frame(self) -> pd.DataFrame:
+        """Generates a dataframe containing all values stored in this class.
+
+        :return: a data frame containing all values stored in this class
+        members
+        """
+        columns = ['Document Name'] + [dictionary.label for dictionary in
+                                       self.get_active_dicts()] + \
+                  ['formula', 'total word count', 'score']
+        indices = [file.label for file in self._corpus] + ['averages']
+        df = pd.DataFrame(columns=range(len(columns)),
```

Comment:
Is it possible that instead of working with **col_index** this Dataframe can work with actual column names? 
`columns=["file_label", ...]`

Reply:
If we do so, how can we then do these lines:
            for col_index, (count) in enumerate(counter):
                df.xs(row_index)[col_index + 1] = count
            df.xs(row_index)[col_index + 2] = formula
            df.xs(row_index)[col_index + 3] = file.total_word_count
            df.xs(row_index)[col_index + 4] = score

Reply:
df.concat

Reply:
Sorry, df.append.